src/pylexibank/dataset.py

```diff
@@ -361,6 +407,194 @@ def coverage(self, vars, glangs, c):  # pragma: no cover
             vars[vid].add(cid)
             glangs[row['Language_ID']].add(cid)
 
+    def build_status_badge(self):
+        if not self.dir.joinpath('.travis.yml').exists():
+            return ''
+        try:
+            return "[![Build Status](https://travis-ci.org/{0}.svg?branch=master)]" \
+                   "(https://travis-ci.org/{0})".format(self.github_repo)
+        except:
+            return ''
+
+    def report(self, tr_analysis, log=None):
+        #
+        # FIXME: write only summary into README.md
+        # in case of multiple cldf datasets:
+        # - separate lexemes.md and transcriptions.md
+        #
+        if not list(self.cldf_dir.glob('*.csv')):
+            return
+        lines = [
+            '# %s\n' % self.metadata.title,
+            'Cite the source dataset as\n',
+            '> %s\n' % self.metadata.citation,
+        ]
+
+        if self.metadata.license:
+            lines.extend([
+                'This dataset is licensed under a %s license' % self.metadata.license, ''])
+
+        if self.metadata.url:
+            lines.extend(['Available online at %s' % self.metadata.url, ''])
+
+        if self.metadata.related:
+            lines.extend(['See also %s' % self.metadata.related, ''])
+
+        if self.metadata.conceptlist:
+            lines.extend([
+                'Conceptlist in Concepticon: [{0}](http://concepticon.clld.org/contributions'
+                '/{0})'.format(self.metadata.conceptlist),
+                ''])
+
+        # add NOTES.md
+        if self.dir.joinpath('NOTES.md').exists():
+            lines.extend(['## Notes', ''])
+            lines.extend(self.dir.joinpath('NOTES.md').read_text().split("\n"))
+            lines.extend(['', ''])  # some blank lines
+
+        synonyms = defaultdict(Counter)
+        totals = {
+            'languages': Counter(),
+            'concepts': Counter(),
+            'sources': Counter(),
+            'cognate_sets': Counter(),
+            'lexemes': 0,
+        }
+
+        missing_source = []
+        missing_lang = [L['NAME'] for L in self.languages if not L['GLOTTOCODE']]
+        missing_param = []
+        missing_concept = [c for c in self.concepts if not c['CONCEPTICON_ID']]
```

Comment:
Hmm. For the missing_{source,param,concept}, would it be quicker to get this from the FormTable, since we're already iterating over everything in that? (rather than iterating over two other files?)

Reply:
It would also be more accurate, because what's measured here is simply how complete the mapping in `etc/concepts.csv` is :)

Reply:
Actually, `missing_param` and `missing_concept` don't make too much sense. The former, because that can never happen - or at least will already result in an invalid CLDF dataset, with errors flagged by Travis. The latter, because we already have the "concepticon mapping"-badge, and figuring out which ones are not mapped is a simple 
```
$ csvgrep -c Concepticon_ID -r"[^0-9]+" cldf/parameters.csv 
```
away.

Reply:
The same could probably be said about `missing_lang`. But for the languages, the "Improvements" section does add value by checking against Glottolog bookkeeping languoids.

Reply:
The main reasons to have these are: 1. to indicate dataset quality & 2. to enable improvement (oh, xx languages aren't mapped? let me fix that).

In that case missing_param is worthless. Is missing_concept a motivator of type 2? (xx items are in concepticon?! let me fix that)? Perhaps not as readily (it also really doesn't help for things like IDS where there's many many items that aren't mapped, and never will be).

Are there any other type 2 statistics we're missing?

Reply:
Thinking more about this -- missing_concept should go, as anything missing is going to stay missing because datasets should be mapped to concepticon as part of adding them to lexibank.

Reply:
@SimonGreenhill right! In the case of concepts, any issues with incomplete mappings should ideally be delegated to corresponding issues with the respective concept list in Concepticon.